### ISSUE TITLE
How to use hub.KerasLayer without the Sequential API

### ISSUE THREAD

**Issue by benderama3** [tensorflow/tensorflow#34053]
I don't know if its a bug.
I'm following this tutorial https://www.tensorflow.org/tutorials/keras/text_classification_with_hub which is working fine.
I'm trying to replace the sequential model: 
```
model = tf.keras.Sequential()
model.add(hub_layer)
model.add(layers.Dense(16, activation='relu'))
model.add(layers.Dense(1, activation='sigmoid'))
```
by:

```
class model(tf.keras.Model):
    def __init__(self, hub_layer):
        super().__init__()
        self.embedding = hub_layer
        self.dense1 = layers.Dense(16, activation='relu')
        self.dense2 = layers.Dense(1, activation='sigmoid')
    def call(self, x):
        x = self.embedding(x)
        x = self.dense1(x)
        return self.dense2(x)
```
Which is throwing:
```
ValueError: Python inputs incompatible with input_signature:
      inputs: (
        Tensor("input_1_35:0", shape=(None, 1), dtype=string))
      input_signature: (
        TensorSpec(shape=(None,), dtype=tf.string, name=None))
```
I didn't change anything else.
If this a bug or I'm doing something wrong ? The second model should act exactly like the first one.

**Comment by gadagashwini** [comment]
Issue is replicating with Tf 2.0.
Please see the gist [here](https://colab.sandbox.google.com/gist/gadagashwini/5ece921ce20ee180da6a24cca64d4bcc/text_classification_with_hub.ipynb). Thanks!

**Comment by ymodak** [comment]
This seems to be fixed with TF  '2.1.0-rc1'. You may try above gist. Thanks!

**Comment by jvishnuvardhan** [comment]
I am closing this issue as it was resolved with recent `tf-nightly`. Please feel free to reopen if the issue persists again. Thanks!
